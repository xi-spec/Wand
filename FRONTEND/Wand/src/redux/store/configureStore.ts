import reduxImmutableStatInvariant from 'redux-immutable-state-invariant';
import thunk from 'redux-thunk';
import { applyMiddleware, compose, createStore } from 'redux';
import rootReducer from '../reducers';

function configureStore () {
  const composeEnhancers = compose;

  return createStore(
    rootReducer,
    composeEnhancers(applyMiddleware(reduxImmutableStatInvariant(), thunk))
  );
}

export default configureStore();
