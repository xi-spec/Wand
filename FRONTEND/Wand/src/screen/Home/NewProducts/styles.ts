
import { StyleSheet } from 'react-native';
import { SIZE, FONTS, COLORS } from '../../../constants/theme';

const styles = StyleSheet.create({
  container: {
    paddingTop: 30,
    margin: 10,
    paddingBottom: 50
  },
  text: {
    fontSize: SIZE.primary,
    margin: 10,
    fontFamily: FONTS.logoRegular
  },
  imageContainer: {
    marginLeft: 5,
    marginRight: 5
  },
  image: {
    justifyContent: 'center',
    width: 170,
    height: 250,
    borderRadius: 5

  },
  detail: {
    fontFamily: FONTS.light,
    fontSize: SIZE.second
  },
  cost: {
    fontFamily: FONTS.primary,
    fontSize: SIZE.second,
    color: COLORS.second
  }

});

export default styles
;
