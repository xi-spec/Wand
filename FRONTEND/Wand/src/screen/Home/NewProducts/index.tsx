import React from 'react';
import {
  Text,
  TouchableOpacity,
  View,
  ScrollView,
  Image
} from 'react-native';
import { connect } from 'react-redux';
import styles from './styles';
import { NavigationScreenProp, NavigationRoute } from 'react-navigation';
import { Params } from '../../../models';
import { Product } from '../../../models/initiaState';
import setCostValue2Decimals from '../../../utils/setCostValue2Decimals';

function NewProducts ({ allProducts, navigation }:{
  navigation: NavigationScreenProp<NavigationRoute<Params>, Params>, allProducts:Product[]}) {
  const randNum = Math.floor(Math.random() * 14);
  const newPrdocuts = allProducts.slice(randNum, randNum + 6);

  return (
    <View style={styles.container}>
   <Text style={styles.text}>
     New arrivals
   </Text>
    <ScrollView
    bounces={false}
    horizontal={true}
    showsHorizontalScrollIndicator={false} >
   {newPrdocuts && newPrdocuts.map((product) => (
     <View
     key={Math.random() * 10}>

    <TouchableOpacity
    style={styles.imageContainer}

    testID='newProduts'
    onPress={
      () =>
        navigation.navigate('Detail', { productId: product._id })
    }>
  <Image source={{ uri: product.images.alls[0] }}
  style={styles.image}>
    </Image>
    <Text style={styles.detail}>{product.name}</Text>
    <Text style={styles.cost}>{`${setCostValue2Decimals(product.cost.value)} ${product.cost.currency}`}</Text>
  </TouchableOpacity>
     </View>

   ))}
  </ScrollView>
  </View>

  );
}

function mapStateToProps ({ products: { allProducts } }) {
  return { allProducts };
}

export default connect(mapStateToProps, null)(NewProducts);
