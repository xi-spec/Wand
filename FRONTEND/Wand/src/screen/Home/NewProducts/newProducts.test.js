
import React from 'react';
import NewProducts from './index';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { render, fireEvent, cleanup } from '@testing-library/react-native';

describe('Given a NewProducts component', () => {
  const mockStore = configureStore([thunk]);
  const store = mockStore({
    products: {
      allProducts: [
        {
          name: 'a',
          images: { alls: ['2', '0'] },
          cost: { value: 1, currency: 'e' }
        },
        {
          name: 'a',
          images: { alls: ['2', '0'] },
          cost: { value: 1, currency: 'e' }
        },
        {
          name: 'a',
          images: { alls: ['2', '0'] },
          cost: { value: 1, currency: 'e' }
        },
        {
          name: 'a',
          images: { alls: ['2', '0'] },
          cost: { value: 1, currency: 'e' }
        },
        {
          name: 'a',
          images: { alls: ['2', '0'] },
          cost: { value: 1, currency: 'e' }
        },
        {
          name: 'a',
          images: { alls: ['2', '0'] },
          cost: { value: 1, currency: 'e' }
        },
        {
          name: 'a',
          images: { alls: ['2', '0'] },
          cost: { value: 1, currency: 'e' }
        },
        {
          name: 'a',
          images: { alls: ['2', '0'] },
          cost: { value: 1, currency: 'e' }
        },
        {
          name: 'a',
          images: { alls: ['2', '0'] },
          cost: { value: 1, currency: 'e' }
        },
        {
          name: 'a',
          images: { alls: ['2', '0'] },
          cost: { value: 1, currency: 'e' }
        },
        {
          name: 'a',
          images: { alls: ['2', '0'] },
          cost: { value: 1, currency: 'e' }
        },
        {
          name: 'a',
          images: { alls: ['2', '0'] },
          cost: { value: 1, currency: 'e' }

        },
        {
          name: 'a',
          images: { alls: ['2', '0'] },
          cost: { value: 1, currency: 'e' }
        },
        {
          name: 'a',
          images: { alls: ['2', '0'] },
          cost: { value: 1, currency: 'e' }
        }
      ]
    }
  });
  afterEach(cleanup);

  describe('When is invoked', () => {
    test('Then it is renders correctly', () => {
      const tree = render(<Provider store={store}><NewProducts /></Provider>);
      expect(tree).toMatchSnapshot();
    });
  });
  describe('When is press in testId navigation', () => {
    test('Then navigation.navigate is invoked', () => {
      const navigation = {
        navigate: jest.fn()
      };

      const { getAllByTestId } = render(<Provider store={store}><NewProducts navigation={navigation}/></Provider>);
      const touches = getAllByTestId('newProduts');
      touches.forEach(touch => fireEvent.press(touch));

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });
});
