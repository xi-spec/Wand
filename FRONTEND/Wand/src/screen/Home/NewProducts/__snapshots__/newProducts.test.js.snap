// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a NewProducts component When is invoked Then it is renders correctly 1`] = `
<View
  style={
    Object {
      "margin": 10,
      "paddingBottom": 50,
      "paddingTop": 30,
    }
  }
>
  <Text
    style={
      Object {
        "fontFamily": "RalewayRegular",
        "fontSize": 18,
        "margin": 10,
      }
    }
  >
    New arrivals
  </Text>
  <RCTScrollView
    bounces={false}
    horizontal={true}
    showsHorizontalScrollIndicator={false}
  >
    <View>
      <View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginLeft": 5,
              "marginRight": 5,
              "opacity": 1,
            }
          }
          testID="newProduts"
        >
          <Image
            source={
              Object {
                "uri": "2",
              }
            }
            style={
              Object {
                "borderRadius": 5,
                "height": 250,
                "justifyContent": "center",
                "width": 170,
              }
            }
          />
          <Text
            style={
              Object {
                "fontFamily": "RalewayLight",
                "fontSize": 14,
              }
            }
          >
            a
          </Text>
          <Text
            style={
              Object {
                "color": "#F3603F",
                "fontFamily": "MontserratRegular",
                "fontSize": 14,
              }
            }
          >
            1.00 e
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
